src/main/java/seedu/address/logic/commands/TableCommand.java

```diff
@@ -133,48 +81,7 @@ private SecLevelTableCommandResult executeSecLevel(Model model) {
      * @return SubjectTableCommandResult instance containing column titles and values.
      */
     private SubjectTableCommandResult executeSubject(Model model) {
-        Map<String, Integer> columnValueMapping = new HashMap<>();
-
-        String[] titles = new String[] {Subject.ENG, Subject.CHI, Subject.EMATH, Subject.AMATH,
-                                        Subject.PHY, Subject.CHEMI, Subject.BIO, Subject.GEOG,
-                                        Subject.HIST, Subject.SOC};
-
-        ObservableList<Student> studentList = model.getFilteredPersonList();
-
-        StudentTakesSubjectPredicate takeEngPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.ENG));
-        StudentTakesSubjectPredicate takeChiPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.CHI));
-        StudentTakesSubjectPredicate takeEMathPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.EMATH));
-        StudentTakesSubjectPredicate takeAMathPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.AMATH));
-        StudentTakesSubjectPredicate takePhyPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.PHY));
-        StudentTakesSubjectPredicate takeChemiPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.CHEMI));
-        StudentTakesSubjectPredicate takeBioPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.BIO));
-        StudentTakesSubjectPredicate takeGeogPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.GEOG));
-        StudentTakesSubjectPredicate takeHistPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.HIST));
-        StudentTakesSubjectPredicate takeSocPredicate = new StudentTakesSubjectPredicate(new Subject(Subject.SOC));
-
-        Stream<StudentTakesSubjectPredicate> predicateStream = Stream.of(
-                takeEngPredicate,
-                takeChiPredicate,
-                takeEMathPredicate,
-                takeAMathPredicate,
-                takePhyPredicate,
-                takeChemiPredicate,
-                takeBioPredicate,
-                takeGeogPredicate,
-                takeHistPredicate,
-                takeSocPredicate
-        );
-
-        int[] values = predicateStream
-                .map(p -> studentList.stream().filter(p).count())
-                .mapToInt(Long::intValue)
-                .toArray();
-
-        for (int i = 0; i < titles.length; i++) {
-            columnValueMapping.put(titles[i], values[i]);
-        }
-
-        return new SubjectTableCommandResult(columnValueMapping);
+        return new SubjectTableCommandResult(OrganizeData.bySubject(model));
```

Comment:
Now I see what you're doing here. Good job on abstracting out the organized data and have particular functions for mapping the count of each category.